iProdyHomework/src/homework14/AttandanceLogger.java

```diff
@@ -0,0 +1,87 @@
+package homework14;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class AttandanceLogger {
+    private String timestamp;
+    private String user_id;
+
+    public AttandanceLogger(String timestamp, String user_id) {
+        this.timestamp = timestamp;
+        this.user_id = user_id;
+    }
+
+    public void registerAttendance(Map<String, String> attendanceLogger) {
+        String timestamp = this.timestamp;
+        String user_id = this.user_id;
+        //в качестве ключа помещаем время, а в качестве значения - id, так как, если сделать
+        //наоборот, будут перезаписываться значения (для одного посетителя не получится сохранить
+        //несколько записей времени
+        attendanceLogger.put(timestamp, user_id);
+    }
+
+    public static TreeMap<String, Integer> countAttendance(Map<String, String> attendanceLogger) {
+        //используем TreeMap, чтобы пользователи были отсортированы
+        TreeMap<String, Integer> attendanceCounter = new TreeMap<>();
```

Comment:
Работа с интерфейсами (абстракция), не с реализациями (детали).
```suggestion
        Map<String, Integer> attendanceCounter = new TreeMap<>();
```